### PR TITLE
[sc] enable faucet for all shards

### PR DESCRIPTION
**Pull request by LeoHChen**
Tested using a new devnet. I can request tokens on all 4 shards.
https://0.harmony.one/#/

Signed-off-by: Leo Chen <leo@harmony.one>

### CODE REVIEW

**Comment by LeoHChen** [comment]
Screen Shot 2019-04-15 at 10 32 47 PM

**Comment by mikedoan** [comment]
will review it in 30min

File: node/node.go [hunk 1]
```diff
@@ -267,18 +267,26 @@ func New(host p2p.Host, consensusObj *consensus.Consensus, db ethdb.Database, is
 
 		node.Consensus.VerifiedNewBlock = make(chan *types.Block)
 
+		// Add Faucet contract to all shards
+		if isFirstTime {
+			// Setup one time smart contracts
+			node.AddFaucetContractToPendingTransactions()
+		} else {
+			node.AddContractKeyAndAddress(scFaucet)
```
Comment (mikedoan) [inline node/node.go]: write a comment here and below to clarify what we need to add for all shards here and for beacon chain below.

**Comment by LeoHChen** [comment]
Jenkins, test this please.

**Pull request merged**